src/openms/source/FORMAT/XMLFile.cpp

```diff
@@ -152,21 +154,38 @@ namespace OpenMS
       try
       {
         parser->parse(*source);
-        delete(parser);
+        delete parser;
         delete source;
       }
       catch (const xercesc::XMLException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
+        delete source;
+        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", 
+            String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
       }
       catch (const xercesc::SAXException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("SAXException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
+        delete source;
+        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "",
+            String("SAXException: ") + StringManager().convert(toCatch.getMessage()));
       }
       catch (const XMLHandler::EndParsingSoftly & /*toCatch*/)
       {
-        //nothing to do here, as this exception is used to softly abort the parsing for whatever reason.
+        // nothing to do here except free memory, as this exception is used to
+        // softly abort the parsing for whatever reason.
+        delete parser;
+        delete source;
```

Comment:
@cbielow @mwalzer do you agree with this, I think this is a memory leak if we use the `EndParsingSoftly` to abort parsing but continue with the execution: in this case we should clean up memory as we have a mem leak otherwise